flocker/control/_protocol.py

```diff
@@ -734,6 +749,7 @@ def connectionMade(self):
         AMP.connectionMade(self)
         self.agent.connected(self)
         self._pinger.start(self, PING_INTERVAL)
+        # Send SetNodeEra using value of /proc/sys/kernel/random/boot_id
```

Comment:
Right, initial connection will result in old state being wiped. Might be a race condition there with updates from control service, so yeah, might want some check in the agent itself if the race condition isn't fixable.

It's worth noting that the worse symptom of the race condition, where we use a non-existent path, will be solved separately by FLOC-3254 if we can manage to implement it. My main focus in this branch is the Docker plugin and other REST API users since they are our main use case going forward.